src/utils/compressImages.ts

```diff
@@ -51,16 +53,21 @@ export const compressImage = ({
 
       ctx.drawImage(image, 0, 0, targetResizingWidth, targetResizingHeight);
 
+      // Change the file.name & file.type for converting file type
+      const targetFileType = outputFormat === 'preserve' ? imageFile.type : `image/${outputFormat}`;
+      const originalSubtype = imageFile.type.split('/').pop();
+      const newSubtype = targetFileType.split('/').pop();
+      const newName = imageFile.name.replace(originalSubtype, newSubtype);
```

Comment:
Is this logic intended to change the file extension included in the file name? It seems somewhat unsafe.

Reply:
```js
'png_file.png'.replace('png', 'jpeg')
// 'jpeg_file.png'
```

Reply:
aha... yes, good catch! let me fix it

Reply:
from Copilot
```ts
function replaceExtension(filename, newExtension) {
  const dotIndex = filename.lastIndexOf('.');
  if (dotIndex === -1) {
    // No extension found, return the original filename
    return filename;
  }
  // Replace the old extension with the new one
  return filename.substring(0, dotIndex) + '.' + newExtension;
}
```